lib/pos_blockchain.go

```diff
@@ -507,9 +514,14 @@ func (bc *Blockchain) processOrphanBlockPoS(block *MsgDeSoBlock) error {
 		// As a spam-prevention measure, we just throw away this block and don't store it.
 		return nil
 	}
+	// Get the snapshot global params
+	snapshotGlobalParams, err := utxoView.GetCurrentSnapshotGlobalParamsEntry()
```

Comment:
Oof, this one is tricky. You'll want to use the snapshot entry based on the epoch the orphan block falls in (not the current snapshot global params based on the committed tip). I think you need to move the isProperlyFormedBlock into the if/else blocks above.

Reply:
good catch!